rclcpp/test/rclcpp/test_timer.cpp

```diff
@@ -334,3 +334,30 @@ INSTANTIATE_TEST_SUITE_P(
     return std::string("unknown");
   }
 );
+
+TEST_P(TestTimer, test_timer_triggered_twice) {
+  std::atomic<int> callback_counter{0};
+
+  rclcpp::TimerBase::SharedPtr timer_called_twice;
+  timer_called_twice = test_node->create_wall_timer(
+    0ms,
+    [&callback_counter]() {
+      callback_counter += 1;
+    }, nullptr, 1);
+
+  executor->spin();
+  ASSERT_EQ(1, callback_counter);
+
+  executor->spin();
+  ASSERT_NE(2, callback_counter);
+  ASSERT_TRUE(timer_called_twice->is_canceled());
+
+  timer_called_twice->reset();
+
+  executor->spin();
+  ASSERT_EQ(2, callback_counter);
+
+  executor->spin();
+  ASSERT_NE(3, callback_counter);
```

Comment:
```suggestion
  ASSERT_EQ(2, callback_counter); // not incremented
```
same rationale as above.